sycl/test-e2e/Basic/group_local_memory.cpp

```diff
@@ -1,3 +1,4 @@
+// REQUIRES: usm_shared_allocations
```

Comment:
Changes like this effectively mean that we are potentially reducing validation coverage for a feature, simply because another feature doesn't work. That doesn't look right, especially considering that using USM for kernel-device communication is not the only way and using regular buffer + accessor is actually guaranteed to always work.

I think that instead of doing `REQUIRES`, we should do a rewrite so that tests are always launched and they do not depend on USM shared allocations (unless their purpose is to test that kind of allocations)

Reply:
Your comment about rewriting usm shared allocations with buffers convinced me that my PR is probably not the best way to move forward long term. That said though, this now means this JIRA and the PR will have to be blocked until all those tests are rewritten. I personally think since there are many such tests that it is better to just create JIRAs for these tests by grouping tests in the same directory together and we can probably split these among several people including me to speed it up.